### PR TITLE
docs: Add a specification document

### PR DESCRIPTION
**Pull request by Lathrisk**

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>interestParty
+   </td>
+   <td>No
+   </td>
+   <td>Creates an unknown node if array is empty
+   </td>
+  </tr>
+  <tr>
+   <td>interestParty.describedByPersonStatement
```
Comment (lgs85) [inline docs/spec.md:77]:
```suggestion
   <td>interestedParty.describedByPersonStatement
```
Here and elsewhere, this is how this field is defined in the schema

File: docs/spec.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>subject.describedByEntityStatement
+   </td>
+   <td>(Yes)
+   </td>
+   <td>Must be connected to either a person or an entity
+   </td>
+  </tr>
+  <tr>
+   <td>interestParty
```
Comment (lgs85) [inline docs/spec.md:69]:
```suggestion
   <td>interestedParty
```

File: docs/spec.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>interestParty.describedByPersonStatement
+   </td>
+   <td>No
+   </td>
+   <td>Either person or entity should be supplied otherwise unknown is assumed
+   </td>
+  </tr>
+  <tr>
+   <td>interestParty.describedByEntityStatement
```
Comment (lgs85) [inline docs/spec.md:85]:
```suggestion
   <td>interestedParty.describedByEntityStatement
```

File: docs/spec.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+## Data requirements
+
+The visualiser requires a minimum set of data fields to produce a directed graph. It is not necessary for the supplied JSON data to be valid BODS data, but it must meet these minimum requirements.
+
+This is not a full schema. The following tables represent the opinionated requirements of the visualiser. Any schema fields that are not present in the tables below are not currently used in the generation of graphs by the visualiser tool.
+
+
+### Edges
+
```
Comment (lgs85) [inline docs/spec.md:17]: Suggest adding here something like: 'Edges are drawn using BODS (or BODS-like) [ownership or control statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#ownership-or-control-statement)'

File: docs/spec.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>Takes precedence over min and max values
+   </td>
+  </tr>
+</table>
+
+
+
+### Person Nodes
+
```
Comment (lgs85) [inline docs/spec.md:153]: Suggest adding 'Person nodes are drawn using BODS (or BODS-like) [person statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#person-statement)'

File: docs/spec.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>No flag will be drawn if the country code is missing
+   </td>
+  </tr>
+</table>
+
+
+
+### Entity Nodes
+
```
Comment (lgs85) [inline docs/spec.md:249]: Suggest adding 'Entity nodes are drawn using BODS (or BODS-like) [entity statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#entity-statement)

File: docs/spec.md [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+Refer to the full [BODS documentation](https://standard.openownership.org/en/0.3.0/) for more information (and the associated [github repository](https://github.com/openownership/data-standard)). 
+
+
+
+## Data requirements
+
+The visualiser requires a minimum set of data fields to produce a directed graph. It is not necessary for the supplied JSON data to be valid BODS data, but it must meet these minimum requirements.
+
+This is not a full schema. The following tables represent the opinionated requirements of the visualiser. Any schema fields that are not present in the tables below are not currently used in the generation of graphs by the visualiser tool.
```
Comment (lgs85) [inline docs/spec.md:13]: Should we clarify explicitly that extra fields will be ignored, or is that obvious?

File: docs/spec.md [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>This is the connecting node and is required
+   </td>
+  </tr>
+  <tr>
+   <td>subject.describedByPersonStatement
+   </td>
+   <td>(Yes)
+   </td>
+   <td>Must be connected to either a person or an entity
```
Comment (lgs85) [inline docs/spec.md:57]:
```suggestion
   <td>Must contain a `statementID` from a person node or an entity node in the dataset
```

File: docs/spec.md [hunk 9]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+  </tr>
+  <tr>
+   <td>subject.describedByEntityStatement
+   </td>
+   <td>(Yes)
+   </td>
+   <td>Must be connected to either a person or an entity
```
Comment (lgs85) [inline docs/spec.md:65]:
```suggestion
   <td>Must contain a `statementID` from a person node or an entity node in the dataset
```

File: docs/spec.md [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>nationalities
+   </td>
+   <td>No
+   </td>
+   <td>No flags will be drawn if these data are missing. The visualiser only uses the first flag in the nationalities array.
+   </td>
+  </tr>
+  <tr>
+   <td>nationalities[0].code
```
Comment (lgs85) [inline docs/spec.md:237]:
```suggestion
   <td>nationalities.code
```

File: docs/spec.md [hunk 11]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>No
+   </td>
+   <td>No flag will be drawn if the country code is missing
```
Comment (lgs85) [inline docs/spec.md:241]:
```suggestion
   <td>No flag will be drawn if the country code is missing. The visualiser only uses the first nationality in the nationalities array.
```

File: docs/spec.md [hunk 12]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>Unknown person will be assumed if the personType is missing.
+   </td>
+  </tr>
+  <tr>
+   <td>nationalities
+   </td>
+   <td>No
+   </td>
+   <td>No flags will be drawn if these data are missing. The visualiser only uses the first flag in the nationalities array.
```
Comment (lgs85) [inline docs/spec.md:233]:
```suggestion
   <td>No flags will be drawn if these data are missing. The visualiser only uses the first nationality in the nationalities array.
```

File: docs/spec.md [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>interests.type
+   </td>
+   <td>No
+   </td>
+   <td>No edge drawn if missing. Only shareholding or votingRights shown
+   </td>
+  </tr>
+  <tr>
+   <td>interests.interestLevel (directOrIndirect)
```
Comment (lgs85) [inline docs/spec.md:109]:
```suggestion
   <td>interests.interestLevel (BODS v0.2)/interests.directOrIndirect (BODS v0.3)
```

File: docs/spec.md [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+  <tr>
+   <td>publicListing
+   </td>
+   <td>No
+   </td>
+   <td>Assumes entity is not listed if this field is missing
+   </td>
+  </tr>
+  <tr>
+   <td>incorporatedInJurisdiction [jurisdiction]
```
Comment (lgs85) [inline docs/spec.md:301]:
```suggestion
   <td>incorporatedInJurisdiction (BODS v0.2)/jurisdiction (BODS v0.3)
```

File: docs/spec.md [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+   </td>
+   <td>No
+   </td>
+   <td>No flags are drawn if this field is missing
```
Comment (lgs85) [inline docs/spec.md:305]: I think we need to add a row for `jurisdiction.code` (assuming this is what you use to generate the entity flags) below this as per the nationalities object?

**Review by lgs85 — COMMENTED** [review]
I think that overall this looks really good. Just a few comments and suggestions re field names.

File: docs/spec.md [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+  <tr>
+   <td>incorporatedInJurisdiction (BODS v0.2)/jurisdiction (BODS v0.3)
+   </td>
+   <td>No
+   </td>
+   <td>No flags are drawn if this field is missing.
+   </td>
+  </tr>
+  <tr>
+   <td>jurisdiction.code
```
Comment (Lathrisk) [inline docs/spec.md:308]: I kept this as just BODS v0.3 for brevity, and hopefully the row above makes it clear how to adapt for v0.2 but happy to change if we need greater clarity.

File: docs/spec.md [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+
+## Data requirements
+
+The visualiser requires JSON data with a minimum set of data fields to produce a directed graph. It is not necessary for the supplied JSON data to be valid BODS data, but it must meet these minimum requirements.
+
+This is not a full schema. The following tables represent the opinionated requirements of the visualiser. Any schema fields that are not present in the tables below are not currently used in the generation of graphs by the visualiser tool and will be ignored.
+
+
+### Edges
+'Edges are drawn using BODS (or BODS-like) [ownership or control statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#ownership-or-control-statement)'
```
Comment (lgs85) [inline docs/spec.md:16]:
```suggestion
Edges are drawn using BODS (or BODS-like) [ownership or control statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#ownership-or-control-statement)
```

File: docs/spec.md [hunk 18]
```diff
@@ -0,0 +1,316 @@
...
+   </td>
+   <td>Takes precedence over min and max values
+   </td>
+  </tr>
+</table>
+
+
+
+### Person Nodes
+Suggest adding 'Person nodes are drawn using BODS (or BODS-like) [person statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#person-statement)'
```
Comment (lgs85) [inline docs/spec.md:152]:
```suggestion
Person nodes are drawn using BODS (or BODS-like) [person statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#person-statement)
```

File: docs/spec.md [hunk 19]
```diff
@@ -0,0 +1,316 @@
...
+   </td>
+   <td>No flag will be drawn if the country code is missing. The visualiser only uses the first nationality in the nationalities array.
+   </td>
+  </tr>
+</table>
+
+
+
+### Entity Nodes
+Suggest adding 'Entity nodes are drawn using BODS (or BODS-like) [entity statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#entity-statement).
```
Comment (lgs85) [inline docs/spec.md:248]:
```suggestion
Entity nodes are drawn using BODS (or BODS-like) [entity statements](https://standard.openownership.org/en/0.3.0/schema/schema-browser.html#entity-statement).
```

**Review by lgs85 — APPROVED** [review]
Looks great - just a couple of typos which are my fault

**Pull request merged**